fe/fe-core/src/main/java/com/starrocks/common/IndexParams.java

```diff
@@ -0,0 +1,205 @@
+// Copyright 2021-present StarRocks, Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.starrocks.common;
+
+import com.starrocks.common.InvertedIndexParams.InvertedIndexImpType;
+import com.starrocks.common.io.ParamsKey;
+import com.starrocks.sql.analyzer.SemanticException;
+import com.starrocks.sql.ast.IndexDef.IndexType;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class IndexParams {
+
+    private static final Logger LOG = LogManager.getLogger(IndexParams.class);
+    private final Map<String, IndexParamItem> paramsHolder = new HashMap<>();
+
+    private IndexParams() {
```

Comment:
if this is not changed dynamically, I suggest build these with static functions and immutable map.